### ISSUE TITLE
Limit Clojure's LazySeq output

### ISSUE THREAD

**Issue by diegobfernandez** [Olical/conjure#104]
Hi, I'm experimenting with Conjure and in my current project I have some big LazySeq.

Printing it in lein repl with ultra is fine because it's truncated, but on Conjure it consumes a lot of cpu and memory and never prints anything.

Is there any configuration to control this behaviour? Should LazySeq be truncated?

**Comment by Olical** [comment]
Oh interesting! There's an nREPL setting to apply a hard limit on the number of bytes printed (quota) which I could start using, the hard part is how to let the user request the rest of it (when the size is unknown!) in a nice way.

There's also [clojure.core/*print-length*](https://clojuredocs.org/clojure.core/*print-length*) which may be easier to use and configure from the user's standpoint.

**Comment by Olical** [comment]
```clojure
; --------------------------------------------------------------------------------
; eval (current-form): (set! *print-length* 5)
5
; --------------------------------------------------------------------------------
; eval (current-form): (range 30)
(0 1 2 3 4 ...)
```

So I'm thinking about integrating the nREPL quota to have a hard limit on output as well as print-length, all through config.

My thinking is this: There's a default limit, if you exceed it then Conjure will warn you that it's been exceeded and which option you should set to remove the limit or raise it (if you need to). What do you think? So no special prefixes or commands to request the rest of the data, just "this has been trimmed for reason X, set this option to nil or a higher number to see more"?

You can then always run `,v1` to see the last result so you don't need to recompute it.

**Comment by Olical** [comment]
I've added limits to the depth and length of data structures on the develop branch. Let me know what you think :smile:

You can use `:helpgrep print_options#level` to show the docs on the two options, they default to a depth of 50 and a length of 500 at the moment.

**Comment by diegobfernandez** [comment]
Thank you so much for the fast response.
I just gave it a try and It is great, I can finally use Conjure on this project.

Informing the user the output is trimmed and how to configure it would be great. I don't really need to load more (or the rest) of the data because it's usually just to peek on it.

Thanks for this awesome plugin, really good stuff.

**Comment by Olical** [comment]
Awesome! That's great to hear! And yeah, I'd love to provide that but I
don't think there's any way for me to know it's been elided... maybe I can
check the string for "..." but that could misfire. I'll have a think!

I hope you enjoy :D also if you have any suggestions on what the default
limits should be, let me know. I took a complete guess.

**Comment by Olical** [comment]
Closing this for now, I didn't add the "hey set this to raise the limit" yet because I can't think of a fool proof way to solve it so I'll give it some more hammock. If you `:helpgrep print_options` for now it'll show you everything you can configure.

This is released in `v4.1.0` :smile: :rocket:

Have a great day!

**Comment by diegobfernandez** [comment]
The defaults seems fine and I usually search the output for what I want. Configuring this is not that valuable as the feature itself. Thanks again.
